### PR TITLE
Add back OSX build on travis

### PR DESCRIPTION
**Pull request by apingault**
BEGINRELEASENOTES
- Add OSX build on travis.
  - Only clang, as no root binaries available for gcc.
  - Build last two release of osx (10.12 and 10.13)
  - Builds are allowed to fail to not pollute the travis build history when clang add some fancy new changes to their compiler :)
 - Add `ROOT_VERSION` as env variable in travis.yml
 - Add `FastFinish` option in travis
    - Fails the build as soon as one linux job failed
    - Do not wait for osx build to finish to mark the build as successfull
ENDRELEASENOTES

TODO: 
 - [x] Restore the `WARNING_AS_ERROR` Flags in the cmake configuration once https://github.com/DQM4HEP/dqm4hep-core/pull/55 is merged

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,22 +32,23 @@ addons:
...
+    fi
   - if [[ "${TRAVIS_EVENT_TYPE}" == "cron" && "${CXX}" == "g++"  ]];
     then ./.travis-ci.d/coverity_scan.sh;
     else ./.travis-ci.d/compile_and_test.sh;
     fi
 
 before_install:
   - ./.travis-ci.d/install_dependencies.sh
+  - if [[ "${TRAVIS_OS_NAME}" == "osx"  ]];
+    then brew install mysql;
```
Comment (rete) [inline .travis.yml:43]: No TravisCI addons for that ?

File: source/CMakeLists.txt [hunk 2]
```diff
@@ -49,6 +49,8 @@ if( DQM4HEP_TESTING )
   dqm4hep_add_test( NAME test-xmlparser ARGS "${PROJECT_SOURCE_DIR}/tests/test_xmlparser.xml" )
   
   # Super-custom test
+  set(CUSTOM_TEST_LIBRARY_DIR ${${PROJECT_NAME}_DEPENDS_LIBRARY_DIRS})
```
Comment (rete) [inline source/CMakeLists.txt:52]: Maaaaaaad world
https://www.youtube.com/watch?v=4N3N1MlvVc4

File: .travis.yml (hunk 1 above)
Comment (apingault) [reply .travis.yml:43]: Apparently not: https://docs.travis-ci.com/user/reference/osx/#Homebrew

Comment (rete) [reply .travis.yml:43]: Fair !

File: source/CMakeLists.txt (hunk 2 above)
Comment (apingault) [reply source/CMakeLists.txt:52]: Yeah...took me more than 1hour to find the problem and make this work

**Pull request merged**